provider/identity_provider_resource.go

```diff
@@ -283,6 +310,22 @@ func (r *IdentityProviderResource) Create(ctx context.Context,
 			htpasswdBuilder.Password(state.HTPasswd.Password.Value)
 		}
 		builder.Htpasswd(htpasswdBuilder)
+    case state.Gitlab != nil:
+		builder.Type(cmv1.IdentityProviderType("GitlabIdentityProvider"))
+		gitlabBuilder := cmv1.NewGitlabIdentityProvider()
+		if !state.Gitlab.CA.Null {
+			gitlabBuilder.CA(state.Gitlab.CA.Value)
+		}
+		if !state.Gitlab.ClientID.Null {
+			gitlabBuilder.ClientID(state.Gitlab.ClientID.Value)
+		}
+		if !state.Gitlab.ClientSecret.Null {
```

Comment:
`ClientSecret` a required param, no need to check if it's nil

Reply:
Done, removed check for if it's nil.